### ISSUE TITLE
Unable to restore backup.

### ISSUE THREAD

**Issue by 82280zx** [Ylianst/MeshCentral#1853]
-------- 9/27/2020, 10:45:28 PM ---- 0.6.49 --------

events.js:183
      throw er; // Unhandled 'error' event
      ^

Error: ENOENT: no such file or directory, open '/home/myusername/meshcentral-data/meshcentral-data/agentserver-cert-private.key'
 
I have the zip file but thats what I see on my log when I attempt to restore it. Is there a manual command or a way to fix the pathing on the file directory? I notice it has meshcentral-data twice, my path has only one folder :) .

**Comment by Ylianst** [comment]
Will look into this. For now, you can just unzip the "meshcentral-data" folder from the backup yourself. Nothing special, just stop the server, unzip and start it again.

**Comment by 82280zx** [comment]
I did that but I still show it as if it's a new installation, none of my clients or users are back.

**Comment by Ylianst** [comment]
Oh! I think I understand the problem, but I am going to take a guess at when is going on. Your trying to restore using an auto-backup file? So, you can only restore using the "My Server" tab a backup that you created using the "My Server" tab. This feature is for very basic servers.

If you are using a backup file that was auto-generated in the "MeshCentral-Backups" folder, you going to have to extract it yourself and make sure the files are in the right place. First, remove the "Meshcentral-data" folder completely and run the server once. It will auto-generate a bunch of certificates and files. Then, stop the server and overwrite them using the files in your backup and start the server again.

Also, are you using MongoDB?

**Comment by 82280zx** [comment]
Yea I was able to restore it, somewhere along the lines my MongoDB got screwed up from an update, but I do nightly backups on my servers so I was able to find one that had mongo running properly and was able to export my database. I followed your guide here http://info.meshcentral.com/downloads/MeshCentral2/MeshCentral2UserGuide-0.2.2.pdf and export my database from mongo, and on my current updated system (after I was able to install the latest mongo community db) imported it and it looks like its all working again. Thank you for all you do for all of us! I'm still pretty new in regards to linux but I know my way around enough to make things work a little bit :)
